.github/workflows/pre-release.yml

```diff
@@ -1,35 +1,74 @@
 name: Pre-release [WIP]
 
 on:
-  workflow_dispatch:
-    inputs:
-      commit:
-        description: 'A full-length commit SHA-1 hash'
-        required: true
-      version:
-        description: 'Intended version (e.g. v0.46.3)'
-        required: true
+  workflow_run:
+    workflows: [Build for the official Metabase release]
+    types: [completed]
 
 env:
   MAX_HASH_LENGTH: 8
   CUSTOM_REPO: ${{ secrets.CUSTOM_RELEASE_REPO }}
 
 jobs:
+  release-artifact:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        edition: [oss, ee]
+    outputs:
+      version: ${{ steps.version-properties.outputs.version }}
+      commit: ${{ steps.version-properties.outputs.commit }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Download Metabase ${{ matrix.edition }} artifact
+        run: |
+          declare -a curlHeaders=('-H' "Accept: application/vnd.github+json" '-H' "Authorization: Bearer $GITHUB_TOKEN" '-H' "X-GitHub-Api-Version: 2022-11-28")
+
+          download_link=$(
+            curl -sL \
+              "${curlHeaders[@]}" \
+              "${{ github.event.workflow_run.artifacts_url }}" \
+              | jq -r --arg name "$ARTIFACT_NAME" '.artifacts[] | select(.name == $name) | .archive_download_url')
+
+          curl -sL "${curlHeaders[@]}" $download_link -o mb.zip
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ARTIFACT_NAME: metabase-${{ matrix.edition }}-uberjar
+      - name: Unzip Metabase artifact containing an uberjar
+        run: unzip mb.zip
+      - name: Extract the version.properties file from the JAR
+        run: |
+          jar xf target/uberjar/metabase.jar version.properties
+          mv version.properties resources/
+      - name: Reveal Metabase ${{ matrix.edition }} properties
+        id: version-properties
+        run: |
+          echo "version=$(grep -o '^tag=.*' ./resources/version.properties | cut -d'=' -f2)" >> $GITHUB_OUTPUT
+          echo "commit=$(cat ./COMMIT-ID)" >> $GITHUB_OUTPUT
+        shell: bash
+      - name: Upload Metabase ${{ matrix.edition }} JAR as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: metabase-${{ matrix.edition }}-uberjar
+          path: |
+            ./target/uberjar/metabase.jar
+            ./COMMIT-ID
+            ./SHA256.sum
```

Comment:
Because of this.
https://github.com/metabase/metabase/blob/5202b6a0e61f63fad9b34c08ee656f93f123826d/.github/workflows/pre-release.yml#L134:L137
But we could maybe use API to download it again.

Reply:
ah gotcha. yeah, much easier to get it that way. 👍